core/controlplane/config/templates/cloud-config-worker

```diff
@@ -476,6 +490,16 @@ ssh_authorized_keys:
         WantedBy=kubelet.service
 {{end}}
 write_files:
+{{- if .CustomFiles}}
+  {{- range $w := .CustomFiles}}
+  - path: {{$w.Path}}
+    permissions: {{$w.PermissionsString}}
+    content: |
+      {{- range $l := $w.ContentArray}}
+      {{$l}}
+      {{- end}}
```

Comment:
@jeremyd  Sorry to add a review comment like this after the delay but
How about making this:

```yaml
encoding: gzip+base64
content: {{$w.GzippedBase64Content}}
```

like how other relatively large files(`.pem` and `.pem.enc` for example) are currently written via write_files, so that we won't bloat the userdata size unnecessarily?
FYI, if we go that way, the `gzipcompressor.CompressString()` func would be useful.